File: src/main/java/no/nav/bidrag/beregn/forskudd/rest/BeregnForskuddConfig.java
package no.nav.bidrag.beregn.forskudd.rest;

import no.nav.bidrag.beregn.forskudd.core.ForskuddCore;
import no.nav.bidrag.beregn.forskudd.rest.consumer.SjablonConsumer;
import no.nav.bidrag.commons.ExceptionLogger;
import no.nav.bidrag.commons.web.CorrelationIdFilter;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.client.RestTemplate;

@Configuration
public class BeregnForskuddConfig {

  @Bean
  public ForskuddCore forskuddCore() {
    return ForskuddCore.getInstance();
  }

  @Bean
  public SjablonConsumer sjablonConsumer(@Value("${BIDRAGSJABLON_URL}") String sjablonBaseUrl, RestTemplate restTemplate) {
    return new SjablonConsumer(restTemplate, sjablonBaseUrl);
  }

  @Bean
  public ExceptionLogger exceptionLogger() {
    return new ExceptionLogger(BidragBeregnForskudd.class.getSimpleName());
  }

  @Bean
  public CorrelationIdFilter correlationIdFilter() {
    return new CorrelationIdFilter();
  }

}

File: src/main/kotlin/no/nav/bidrag/beregn/forskudd/rest/BeregnForskuddConfig.kt
package no.nav.bidrag.beregn.forskudd.rest

import io.swagger.v3.oas.annotations.OpenAPIDefinition
import io.swagger.v3.oas.annotations.enums.SecuritySchemeType
import io.swagger.v3.oas.annotations.info.Info
import io.swagger.v3.oas.annotations.security.SecurityRequirement
import io.swagger.v3.oas.annotations.security.SecurityScheme
import no.nav.bidrag.beregn.forskudd.BeregnForskuddApi
import no.nav.bidrag.commons.util.OpenApiExamplesConfig
import no.nav.bidrag.commons.web.CorrelationIdFilter
import no.nav.bidrag.commons.web.DefaultCorsFilter
import no.nav.bidrag.commons.web.UserMdcFilter
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Import

@Configuration
@OpenAPIDefinition(
    info = Info(title = "bidrag-beregn-forskudd-rest", version = "v1"),
    security = [SecurityRequirement(name = "bearer-key")],
)
@SecurityScheme(
    bearerFormat = "JWT",
    name = "bearer-key",
    scheme = "bearer",
    type = SecuritySchemeType.HTTP,
)
@Import(CorrelationIdFilter::class, UserMdcFilter::class, DefaultCorsFilter::class, BeregnForskuddApi::class, OpenApiExamplesConfig::class)
class BeregnForskuddConfig
